plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -350,6 +362,151 @@ public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjecti
                 precalculateStatisticsForPushdown));
     }
 
+    private Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjectionExpressions(
+            ConnectorSession session,
+            JdbcTableHandle handle,
+            List<ConnectorExpression> projections,
+            Map<String, ColumnHandle> assignments)
+    {
+        int nextSyntheticColumnId = handle.getNextSyntheticColumnId();
+
+        ImmutableSet.Builder<JdbcColumnHandle> newColumnsBuilder = ImmutableSet.builder();
+        ImmutableList.Builder<Assignment> assignmentBuilder = ImmutableList.builder();
+        ImmutableMap.Builder<ConnectorExpression, Variable> newVariablesBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<String, ParameterizedExpression> columnExpressionsBuilder = ImmutableMap.builder();
+        Set<ConnectorExpression> translatedExpression = new HashSet<>();
+
+        if (isComplexExpressionPushdown(session)) {
```

Comment:
This could be a first statement in the method, and if false - `return Optional.empty();`

Reply:
I don't think we need to return `Optional#empty` as we need to additional changes for normal expression.

Reply:
`if (!newVariables.isEmpty()) {` is true only if `newVariables` is populated. 
`newVariables` is populated only if `newVariablesBuilder` is populated.
`newVariablesBuilder` is populated only when `if (isComplexExpressionPushdown(session)) {` is true.
What do I miss?

Reply:
Okay !! I get it - Previously I was a bit confused as this code snippet was a part of `applyProjection`. Thanks for pointing it out.